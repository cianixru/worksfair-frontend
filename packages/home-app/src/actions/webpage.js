import axios from 'axios';

import { baseURL } from '../utils/api';

const apiToken = localStorage.getItem('token');

export const NEW_WEBPAGE = 'NEW_WEBPAGE';
export const CREATE_WEBPAGE_FAILED = 'CREATE_WEBPAGE_FAILED';

/**
 * Sends new webpage data to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const createWebpage = (data) => async (dispatch) => {
  try {
    const newWebpage = await axios({
      url: '/webpages/',
      method: 'post',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: NEW_WEBPAGE,
      data: newWebpage.data,
    });
  } catch (error) {
    return dispatch({
      type: CREATE_WEBPAGE_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const GET_WEBPAGE = 'GET_WEBPAGE';
export const GET_WEBPAGE_FAILED = 'GET_WEBPAGE_FAILED';

/**
 * Fetches webpage data from the backend
 * @param { string } data
 * @returns { func } dispatch
 */
export const getWebpage = (data) => async (dispatch) => {
  try {
    const webpage = await axios({
      url: `/webpages/${data}/`,
      method: 'get',
      data,
      baseURL,
    });
    return dispatch({
      type: GET_WEBPAGE,
      data: webpage.data,
    });
  } catch (error) {
    return dispatch({
      type: GET_WEBPAGE_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const UPDATED_WEBPAGE = 'UPDATED_WEBPAGE';
export const UPDATE_WEBPAGE_FAILED = 'UPDATE_WEBPAGE_FAILED';

/**
 * updates webpage data from the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const updateWebpage = (data) => async (dispatch) => {
  try {
    const webpage = await axios({
      url: `/webpages/${data.subDomainName}/`,
      method: 'patch',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: UPDATED_WEBPAGE,
      data: webpage.data,
    });
  } catch (error) {
    return dispatch({
      type: UPDATE_WEBPAGE_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const NEW_OFFERING = 'NEW_OFFERING';
export const CREATE_OFFERING_FAILED = 'CREATE_OFFERING_FAILED';

/**
 * Sends request for a new offering data to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const createWebpageOffering = (data) => async (dispatch) => {
  try {
    const newOffering = await axios({
      url: `/webpages/${data.subDomainName}/offerings/`,
      method: 'post',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: NEW_OFFERING,
      data: newOffering.data,
    });
  } catch (error) {
    return dispatch({
      type: CREATE_OFFERING_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const GET_WEBPAGES = 'GET_WEBPAGES';
/**
 * Gets webpages from data
 * @param { object } data
 * @returns { func } dispatch
 */
export const getWebpages = (data) => (dispatch) => {
  return dispatch({
    type: GET_WEBPAGES,
    data,
  });
};

export const UPDATED_OFFERING = 'UPDATED_OFFERING';
export const UPDATE_OFFERING_FAILED = 'UPDATE_OFFERING_FAILED';
/**
 * Sends request for updating an offering to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const updateOffering = (data) => async (dispatch) => {
  try {
    const updatedOffering = await axios({
      url: `/webpages/${data.subDomainName}/offerings/${data.id}`,
      method: 'patch',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: UPDATED_OFFERING,
      data: updatedOffering.data,
    });
  } catch (error) {
    return dispatch({
      type: UPDATE_OFFERING_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const DELETE_OFFERING = 'DELETE_OFFERING';
export const DELETE_OFFERING_FAILED = 'DELETE_OFFERING_FAILED';
/**
 * Sends request for deleting an offering to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const deleteOffering = (data) => async (dispatch) => {
  try {
    await axios({
      url: `/webpages/${data.subDomainName}/offerings/${data.id}`,
      method: 'delete',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: DELETE_OFFERING,
    });
  } catch (error) {
    return dispatch({
      type: DELETE_OFFERING_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const NEW_DETAIL = 'NEW_DETAIL';
export const CREATE_DETAIL_FAILED = 'CREATE_DETAIL_FAILED';

/**
 * Sends request for a new detail data to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const createWebpageDetails = (data) => async (dispatch) => {
  try {
    const newDetail = await axios({
      url: `/webpages/${data.subDomainName}/details/`,
      method: 'post',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: NEW_DETAIL,
      data: newDetail.data,
    });
  } catch (error) {
    return dispatch({
      type: CREATE_DETAIL_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const UPDATED_DETAIL = 'UPDATED_DETAIL';
export const UPDATE_DETAIL_FAILED = 'UPDATE_DETAIL_FAILED';
/**
 * Sends request for updating an detail to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const updateDetail = (data) => async (dispatch) => {
  try {
    const updatedDetail = await axios({
      url: `/webpages/${data.subDomainName}/details/${data.id}`,
      method: 'patch',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: UPDATED_DETAIL,
      data: updatedDetail.data,
    });
  } catch (error) {
    return dispatch({
      type: UPDATE_DETAIL_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};

export const DELETE_DETAIL = 'DELETE_DETAIL';
export const DELETE_DETAIL_FAILED = 'DELETE_DETAIL_FAILED';
/**
 * Sends request for deleting an detail to the backend
 * @param { object } data
 * @returns { func } dispatch
 */
export const deleteDetail = (data) => async (dispatch) => {
  try {
    await axios({
      url: `/webpages/${data.subDomainName}/details/${data.id}`,
      method: 'delete',
      data,
      baseURL,
      headers: { Authorization: `Token ${apiToken}` },
    });
    return dispatch({
      type: DELETE_DETAIL,
    });
  } catch (error) {
    return dispatch({
      type: DELETE_DETAIL_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};
