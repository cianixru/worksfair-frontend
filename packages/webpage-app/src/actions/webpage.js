import api from '../utils/api';

export const GET_WEBPAGE = 'GET_WEBPAGE';
export const GET_WEBPAGE_FAILED = 'GET_WEBPAGE_FAILED';

/**
 * Fetches webpage data from the backend
 * @param { string } data
 * @returns { func } dispatch
 */
export const getWebpage = data => async (dispatch) => {
  try {
    const webpage = await api.get(`/webpages/${data}/`);
    return dispatch({
      type: GET_WEBPAGE,
      data: webpage.data,
    });
  } catch (error) {
    return dispatch({
      type: GET_WEBPAGE_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};
