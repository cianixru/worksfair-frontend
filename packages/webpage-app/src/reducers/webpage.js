import {
  GET_WEBPAGE,
  GET_WEBPAGE_FAILED,
} from '../actions/webpage';

const initialState = {
  webpage: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
  case GET_WEBPAGE:
    return { ...state, webpage: action.data };

  case GET_WEBPAGE_FAILED:
    return { ...state, webpage: null };

  default:
    return state;
  }
};
