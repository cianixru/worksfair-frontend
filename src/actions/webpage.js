import api from '../utils/api';

const apiToken = localStorage.getItem('token');
api.defaults.headers.common.Authorization = `Token ${apiToken}`;

export const NEW_WEBPAGE = 'NEW_WEBPAGE';
export const CREATE_WEBPAGE_FAILED = 'CREATE_WEBPAGE_FAILED';
export const createWebpage = data => async (dispatch) => {
  try {
    const newWebpage = await api.post('/webpages/', data);
    return dispatch({
      type: NEW_WEBPAGE,
      data: newWebpage.data,
    });
  } catch (error) {
    return dispatch({
      type: CREATE_WEBPAGE_FAILED,
      message: error.message,
      response: error.response,
    });
  }
};
