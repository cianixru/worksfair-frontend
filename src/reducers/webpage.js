import {
  NEW_WEBPAGE,
  CREATE_WEBPAGE_FAILED,
  GET_WEBPAGE,
  GET_WEBPAGE_FAILED,
} from '../actions/webpage';

const initialState = {
  newWebpage: {},
  webpage: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
  case NEW_WEBPAGE:
    return { ...state, newWebpage: action.data };

  case CREATE_WEBPAGE_FAILED:
    return { ...state, newWebpage: null };

  case GET_WEBPAGE:
    return { ...state, webpage: action.data };

  case GET_WEBPAGE_FAILED:
    return { ...state, webpage: null };
  default:
    return state;
  }
};
