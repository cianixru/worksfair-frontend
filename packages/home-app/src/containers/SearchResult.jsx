import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import
WebpageItem from '../components/Dashboard/SearchWebpage/SearchWebpageItem';
import
NoSearchResults from '../components/Dashboard/SearchWebpage/NoSearchResult';
import { search } from '../actions/public';
import SearchForm from '../forms/Public/SearchForm';
import { isLoading, isComplete } from '../actions/loader';

class SearchResultContainer extends Component {
  static propTypes = {
    webpages: PropTypes.array,
    location: PropTypes.object,
    actions: PropTypes.object,
    history: PropTypes.object,
    search: PropTypes.func,
  }

  async componentDidMount() {
    const { location, actions } = this.props;
    try {
      actions.isLoading();
      const params = new URLSearchParams(location.search);
      const data = {
        keywords: params.get('query'),
        location: params.get('location'),
      };
      await actions.search(data);
    } catch (error) {
      console.log(error);
    } finally {
      actions.isComplete();
    }
  }

  onSearch = async (data) => {
    if (!data.location) {
      data.location = 'Nigeria';
    }
    const { history, actions } = this.props;
    try {
      actions.isLoading();
      history.push(`/search?query=${data.keywords}&location=${data.location}`);
      await actions.search(data);
    } catch (error) {
      console.log(error);
    } finally {
      actions.isComplete();
    }
  }

  render() {
    const { webpages, location } = this.props;
    const params = new URLSearchParams(location.search);
    const data = {
      keywords: params.get('query'),
      location: params.get('location'),
    };
    const sampleImage = "https://imgplaceholder.com/180x180/131111?text=PICTURE+COMING+SOON&font-size=18"

    return (
      <section className="section">
        <Helmet>
          <title>Search - Worksfair</title>
        </Helmet>
        <div className="container margin-bottom-25">
          <div className="columns">
            <div className="column is-1" />
            <div className="column">
              <SearchForm
                onSubmit={this.onSearch}
                data={data}
              />
            </div>
            <div className="column is-1" />
          </div>
        </div>
        <div className="container">
          <div className="columns">
            <div className="column is-1" />
            <div className="column">
              <ul>
                { webpages && webpages.length > 0
                  ? webpages.map((webpage) => {
                    const dayCreated = new Date(webpage.created_at).toDateString();

                    const itemProps = {
                      dayCreated, webpage, sampleImage,
                    };
                    return (
                      <WebpageItem key={webpage.sub_domain_name} {...itemProps} />
                    );
                  })
                  : <NoSearchResults />
                }
              </ul>
            </div>
            <div className="column is-1" />
          </div>
        </div>
      </section>
    );
  }
}

const mapStateToProps = ({ publicData: { webpages } }) => ({
  webpages,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      search,
      isComplete,
      isLoading,
    },
    dispatch,
  ),
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchResultContainer));
