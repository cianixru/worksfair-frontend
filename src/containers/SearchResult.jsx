import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import
WebpageItem from '../components/Dashboard/SearchWebpage/SearchWebpageItem';
import
NoSearchResults from '../components/Dashboard/SearchWebpage/NoSearchResult';
import sampleImage from '../assets/sample-01.jpg';
import { search } from '../actions/public';
import SearchForm from '../forms/Public/SearchForm';
import { isLoading, isComplete } from '../actions/loader';

class SearchResultContainer extends Component {
  static propTypes = {
    webpages: PropTypes.array,
    location: PropTypes.object,
    actions: PropTypes.object,
    history: PropTypes.object,
    search: PropTypes.func,
  }

  async componentDidMount() {
    const { location, actions } = this.props;
    try {
      actions.isLoading();
      const params = new URLSearchParams(location.search);
      const data = {
        keywords: params.get('query'),
        location: params.get('location'),
      };
      await actions.search(data);
    } catch (error) {
      console.log(error);
    } finally {
      actions.isComplete();
    }
  }

  onSearch = async (data) => {
    if (!data.location) {
      data.location = 'Nigeria';
    }
    const { history, actions } = this.props;
    try {
      actions.isLoading();
      history.push(`/search?query=${data.keywords}&location=${data.location}`);
      await actions.search(data);
    } catch (error) {
      console.log(error);
    } finally {
      actions.isComplete();
    }
  }

  render() {
    const { webpages, location } = this.props;
    const params = new URLSearchParams(location.search);
    const data = {
      keywords: params.get('query'),
      location: params.get('location'),
    };
    return (
      <section className="section">
        <Helmet>
          <title>Search - Worksfair</title>
        </Helmet>
        <div className="container margin-bottom-25">
          <SearchForm
            onSubmit={this.onSearch}
            data={data}
          />
        </div>
        <div className="container">
          <ul>
            { webpages && webpages.length > 0
              ? webpages.map((webpage) => {
                const dayCreated = new Date(webpage.created_at).toDateString();

                const itemProps = {
                  dayCreated, webpage, sampleImage,
                };
                return (
                  <WebpageItem key={webpage.sub_domain_name} {...itemProps} />
                );
              })
              : <NoSearchResults />
            }
          </ul>
        </div>
      </section>
    );
  }
}

const mapStateToProps = ({ publicData: { webpages } }) => ({
  webpages,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      search,
      isComplete,
      isLoading,
    },
    dispatch,
  ),
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchResultContainer));
