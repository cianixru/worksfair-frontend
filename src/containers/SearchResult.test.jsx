import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import SearchResult from './SearchResult';
import { webpage } from '../actions/webpage.test';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);

describe('SearchResult.jsx', () => {
  test('Should display the SearchResult with no results', () => {
    const mockStore = store({
      auth: { currentUser: {} },
      publicData: { webpages: [] }
    });

    const component = render(
      <Provider store={mockStore}>
        <Router>
          <SearchResult />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
  test('Should display the SearchResult Component with results', () => {
    const mockStore = store({
      auth: { currentUser: {} },
      publicData: { webpages: [webpage] }
    });

    const component = render(
      <Provider store={mockStore}>
        <Router>
          <SearchResult />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
