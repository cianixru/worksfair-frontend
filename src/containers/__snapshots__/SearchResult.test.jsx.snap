// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult.jsx Should display the SearchResult Component with results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section"
      >
        <div
          class="container margin-bottom-25"
        >
          <form>
            <div
              class="main-search-input"
            >
              <div
                class="main-search-input-item"
              >
                <input
                  class="input"
                  name="keywords"
                  placeholder="What do you need?"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="main-search-input-item location"
              >
                <input
                  class="input"
                  name="location"
                  placeholder="Which city or town?"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="button is-info is-rounded"
                disabled=""
              >
                Search
              </button>
            </div>
          </form>
        </div>
        <div
          class="container"
        >
          <ul>
            <li>
              <div
                class="columns box"
              >
                <div
                  class="column is-3"
                >
                  <figure
                    class="image is-180x180"
                  >
                    <img
                      alt="sample"
                      src="sample-01.jpg"
                    />
                  </figure>
                </div>
                <div
                  class="column is-7"
                >
                  <div
                    class="content"
                  >
                    <h3>
                      <a
                        class="has-text-grey"
                        href="/webpage/ideosynergy"
                      >
                        Theookafor2
                      </a>
                    </h3>
                    <p>
                      eod
                    </p>
                    <p
                      class="is-size-6"
                    >
                      undefined, 
                      <strong>
                        undefined, undefined 
                      </strong>
                    </p>
                  </div>
                  <div>
                    <h6
                      class="is-size-7"
                    >
                      <strong>
                        Added to Worksfair on:
                      </strong>
                      <span
                        class="has-text-black"
                      >
                         
                        Invalid Date
                      </span>
                    </h6>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section"
    >
      <div
        class="container margin-bottom-25"
      >
        <form>
          <div
            class="main-search-input"
          >
            <div
              class="main-search-input-item"
            >
              <input
                class="input"
                name="keywords"
                placeholder="What do you need?"
                type="text"
                value=""
              />
            </div>
            <div
              class="main-search-input-item location"
            >
              <input
                class="input"
                name="location"
                placeholder="Which city or town?"
                type="text"
                value=""
              />
            </div>
            <button
              class="button is-info is-rounded"
              disabled=""
            >
              Search
            </button>
          </div>
        </form>
      </div>
      <div
        class="container"
      >
        <ul>
          <li>
            <div
              class="columns box"
            >
              <div
                class="column is-3"
              >
                <figure
                  class="image is-180x180"
                >
                  <img
                    alt="sample"
                    src="sample-01.jpg"
                  />
                </figure>
              </div>
              <div
                class="column is-7"
              >
                <div
                  class="content"
                >
                  <h3>
                    <a
                      class="has-text-grey"
                      href="/webpage/ideosynergy"
                    >
                      Theookafor2
                    </a>
                  </h3>
                  <p>
                    eod
                  </p>
                  <p
                    class="is-size-6"
                  >
                    undefined, 
                    <strong>
                      undefined, undefined 
                    </strong>
                  </p>
                </div>
                <div>
                  <h6
                    class="is-size-7"
                  >
                    <strong>
                      Added to Worksfair on:
                    </strong>
                    <span
                      class="has-text-black"
                    >
                       
                      Invalid Date
                    </span>
                  </h6>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchResult.jsx Should display the SearchResult with no results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section"
      >
        <div
          class="container margin-bottom-25"
        >
          <form>
            <div
              class="main-search-input"
            >
              <div
                class="main-search-input-item"
              >
                <input
                  class="input"
                  name="keywords"
                  placeholder="What do you need?"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="main-search-input-item location"
              >
                <input
                  class="input"
                  name="location"
                  placeholder="Which city or town?"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="button is-info is-rounded"
                disabled=""
              >
                Search
              </button>
            </div>
          </form>
        </div>
        <div
          class="container"
        >
          <ul>
            <li>
              <div
                class="message is-warning"
              >
                <div
                  class="message-body"
                >
                  <p
                    class="is-size-5 margin-bottom-25"
                  >
                    Sorry, we could not find a webpage matching your description.
                  </p>
                  <p
                    class="has-text-info"
                  >
                    Perhaps you can try better keywords
                  </p>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section"
    >
      <div
        class="container margin-bottom-25"
      >
        <form>
          <div
            class="main-search-input"
          >
            <div
              class="main-search-input-item"
            >
              <input
                class="input"
                name="keywords"
                placeholder="What do you need?"
                type="text"
                value=""
              />
            </div>
            <div
              class="main-search-input-item location"
            >
              <input
                class="input"
                name="location"
                placeholder="Which city or town?"
                type="text"
                value=""
              />
            </div>
            <button
              class="button is-info is-rounded"
              disabled=""
            >
              Search
            </button>
          </div>
        </form>
      </div>
      <div
        class="container"
      >
        <ul>
          <li>
            <div
              class="message is-warning"
            >
              <div
                class="message-body"
              >
                <p
                  class="is-size-5 margin-bottom-25"
                >
                  Sorry, we could not find a webpage matching your description.
                </p>
                <p
                  class="has-text-info"
                >
                  Perhaps you can try better keywords
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
